Issue 121 - Feat: Completed Favorites page

rickansen opened the pull request:
## Describe your changes

- Added Favorites route
- Implemented favorite context
- If there is no item in favoriteContext, it will show a different page
- Updated header component with the favorites/heart icon
- Favorites/Heart Icon is now a component to be abstracted from the header component 
- Features of new favorite component inside header component:
     - if favorite object does not the product id 
          - Can add product to favoritesContext onClick
          - Will change the SVG color fill 
          
      - if favorite object already have the product id 
          - Can remove product to favoritesContext onClick
          - Will change the SVG color back to its original color

## Issue ticket number and link
Resolves #121 

## Name of Contributor
Andersen

## Type of change
Feature

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My code does not generate any new warnings or errors
- [ ] New and existing unit tests pass locally with my changes (if relevant)

dmitrykulakovfrontend reviewed (changes requested):
Amazing job, just needs a small change :)

rickansen commented:
Went out of scope with the issue and refactored Items page and its updateBasket function. I refactored it to create new array to set the state instead of mutating the original array.

dmitrykulakovfrontend reviewed (approved):
Awesome!
